### PR TITLE
modal window

### PR DESCRIPTION
**Pull request by AegisVP**
ModalWindow has the following methods:
`.show(modalContentsHtml)` - show a modal window with the passed contents
`.update(modalContentsHtml)` - update an already open modal with the passed contents
`.close()` - close an open modal window

### CODE REVIEW

**Comment by AegisVP** [comment]
Don't merge, will change to singleton...

File: src/js/utils/modalWindow.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  constructor(modalRootElement, options = {}) {
+    this.backdropElement = modalRootElement;
+
+    for (const [key, value] of Object.entries(options)) this[key] = value;
+  }
+
+  show(modalContents) {
+    if ('' === modalContents) return this.close();
+    if (!'' === this.modalContents) return this.update(modalContents);
```
Comment (yevhen-zakharevych) [inline src/js/utils/modalWindow.js:21]: does this condition work?
`!''` is always true
`this.modalContents` string?

Comment (AegisVP) [reply src/js/utils/modalWindow.js:21]: This is **yoda notation**, same thing as `(!this.modalContents === '')`. Protects from inadvertent assignments like this: `(!this.modalContents = '')`

**Comment by AegisVP** [comment]
> Don't merge, will change to singleton...

Not worth it. Can be merged, if needed

**Pull request closed without merging**